### ISSUE TITLE
Errors executing Debug-ContainerHost.ps1

### ISSUE THREAD

**Issue by WCLucas42** [docker/for-win#1719]
### Expected behavior
Running Debug-ContainerHost.ps1 completes without error
### Actual behavior
What I am attempting seems to be working but the script returns the following errors. 

I am just getting started here and want to make sure I am on sound footing before progressing much further.

PS C:\WINDOWS\system32> Invoke-WebRequest https://aka.ms/Debug-ContainerHost.ps1 -UseBasicParsing | Invoke-Expression
Checking for common problems
Container Host OS Product Name: Windows 10 Pro
Container Host OS Build Label: 16299.15.amd64fre.rs3_release.170928-1534
Describing Windows Version and Prerequisites
 [+] Is Windows 10 Anniversary Update or Windows Server 2016 112ms
 [+] Has KB3192366, KB3194496, or later installed if running Windows build 14393 11m
s
 [+] Is not a build with blocking issues 8ms
 [+] Has 'Containers' feature installed 11.94s
Describing Docker is installed
 [+] A Docker service is installed - 'Docker' or 'com.Docker.Service'  28ms
 [+] Service is running 8ms
 [+] Docker.exe is in path 2.07s
 [+] Docker is registered in the EventLog service 13ms
Describing User has permissions to use Docker daemon
 [+] docker.exe should not return access denied 20ms
Describing Windows container settings are correct
 [+] Do not have DisableVSmbOplock set to 1 23ms
 [+] Do not have zz values set 22ms
 [+] Do not have FDVDenyWriteAccess set to 1 45ms
Describing The right container base images are installed
 [+] At least one of 'microsoft/windowsservercore' or 'microsoft/nanoserver' should 
be installed 121ms
Describing Container network is created
 [+] At least one local container network is available 2.01s
 [+] At least one NAT, Transparent, or L2Bridge Network exists 8ms
 [+] NAT Network's vSwitch is internal 7ms
 [-] A Windows NAT is configured if a Docker NAT network exists 81ms
   Expected {0} to be greater than or equal to {1}
   226:        $winnatCount | Should Not BeLessThan $natCount
   at <ScriptBlock>, <No file>: line 226
 [-] Specified Network Gateway IP for NAT network is assigned to Host vNIC 38ms
   Expected: {True}
   But was:  {False}
   240:       $vmNicGatewayIPExists | Should Be $true
   at <ScriptBlock>, <No file>: line 240
 [-] NAT Network's internal prefix does not overlap with external IP' 34ms
   Expected: value was {}, but should not have been the same
   252:              $BinaryIP.Substring(0, $Length) | Should Not Be $BinaryIPSubnet
.Substring(0, $Length)
   at <ScriptBlock>, <No file>: line 252
   at <ScriptBlock>, <No file>: line 247
Showing output from: docker info
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 14
Server Version: 17.12.0-ce
Storage Driver: windowsfilter
 Windows: 
Logging Driver: json-file
Plugins:
 Volume: local
 Network: ics l2bridge l2tunnel nat null overlay transparent
 Log: awslogs etwlogs fluentd gelf json-file logentries splunk syslog
Swarm: inactive
Default Isolation: hyperv
Kernel Version: 10.0 16299 (16299.15.amd64fre.rs3_release.170928-1534)
Operating System: Windows 10 Pro
OSType: windows
Architecture: x86_64
CPUs: 8
Total Memory: 15.91GiB
Name: XXXXXX
ID: QOKG:TSAT:VI7Y:DZXC:L3M5:7U5D:XEHB:M6C6:FCSG:J3ZH:3DWM:FK3S
Docker Root Dir: C:\ProgramData\Docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: -1
 Goroutines: 43
 System Time: 2018-02-16T21:51:41.4970077-05:00
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

Showing output from: docker version
Client:
 Version:	17.12.0-ce
 API version:	1.35
 Go version:	go1.9.2
 Git commit:	c97c6d6
 Built:	Wed Dec 27 20:05:22 2017
 OS/Arch:	windows/amd64

Server:
 Engine:
  Version:	17.12.0-ce
  API version:	1.35 (minimum version 1.24)
  Go version:	go1.9.2
  Git commit:	c97c6d6
  Built:	Wed Dec 27 20:15:52 2017
  OS/Arch:	windows/amd64
  Experimental:	true

Showing output from: docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
4df8b31fb0fb        Default Switch      ics                 local
0a0817458928        nat                 nat                 local
31c24d976291        none                null                local

Getting Warnings & errors in the Windows event logs from the last 24 hours

   ProviderName: docker

TimeCreated                     Id LevelDisplayName Message                        
-----------                     -- ---------------- -------                        
2/16/2018 8:55:48 PM             1 Error            Handler for POST /v1.35/cont...
2/16/2018 8:55:48 PM             1 Error            be01534f209c829158b64911d45e...
2/16/2018 8:00:41 PM             1 Error            Handler for POST /v1.35/buil...
2/16/2018 7:59:53 PM            11 Warning          unable to get exit code for ...
2/16/2018 9:43:48 AM             1 Error            Handler for POST /v1.35/buil...

### Information
Ran the script with instructions from this page
https://docs.microsoft.com/en-us/virtualization/windowscontainers/troubleshooting

I read through these two issues
https://github.com/docker/for-win/issues/294
https://github.com/MicrosoftDocs/Virtualization-Documentation/pull/513

running docker network inspect 0a0817458928 returns:

PS C:\WINDOWS\system32> docker network inspect 0a0817458928
[
    {
        "Name": "nat",
        "Id": "0a081745892887a571cc856fcfdad1b04d2f0b4ae763de94d2036109fad44a69",
        "Created": "2018-02-15T17:01:24.7270646-05:00",
        "Scope": "local",
        "Driver": "nat",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "windows",
            "Options": null,
            "Config": [
                {
                    **"Subnet": "0.0.0.0/0",
                    "Gateway": "0.0.0.0"**
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "bf968f03e1ec5e225803690b82687e1bdfa4ecd10c7f719d1524e9938bdc4bec": {
                "Name": "aspnetcore_sample",
                "EndpointID": "dd5bd7c805c92522c874a6da6cc42826e8a7a125f465faa8a4f42
25805b2aa4e",
                "MacAddress": "00:15:5d:dc:26:5b",
                "IPv4Address": "172.17.199.49/16",
                "IPv6Address": ""
            }
        },
        "Options": {
            "com.docker.network.windowsshim.hnsid": "8cd02cc8-3a88-4ac4-8cde-c478e0b
a1bef",
            "com.docker.network.windowsshim.networkname": "nat"
        },
        "Labels": {}
    }
]

running docker run --rm -ti microsoft/nanoserver ping 8.8.8.8 successfully pings the external address

using the aspnetcore_sample container

docker ps shows:

PS C:\WINDOWS\system32> docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED              STATUS              PORTS                  NAMES
3fe27fb96b44        aspnetapp           "dotnet aspnetapp.dll"   About a minute ago   Up About a minute   0.0.0.0:5000->80/tcp   aspnetcore_sample

running docker exec aspnetcore_sample ipconfig rerturns:

Windows IP Configuration

Ethernet adapter Ethernet:

   Connection-specific DNS Suffix  . : hsd1.pa.comcast.net.
   Link-local IPv6 Address . . . . . : fe80::9c6c:600b:9760:b7a%4
   IPv4 Address. . . . . . . . . . . : 172.17.195.110
   Subnet Mask . . . . . . . . . . . : 255.255.240.0
   Default Gateway . . . . . . . . . : 172.17.192.1

browsing http://172.17.195.110/ displays the aspnetapp webpage

but looking in the log the following is repeated about every 20 seconds while the container is running

[22:09:20.507][WindowsDaemon  ][Info   ] debug: Calling GET /v1.34/containers/json?limit=0
[22:09:20.507][LocalhostForwarder][Info   ] TransortErrorException 1

[22:09:20.508][LocalhostForwarder][Info   ] Failed to forward tcp port 5000 to 172.17.195.110:80
[22:09:20.508][LocalhostForwarder][Info   ] Reconciliation failed: invalid argument
[22:09:20.508][WindowsDaemon  ][Info   ] debug: Calling GET /v1.34/containers/3fe27fb96b442705841ef43d572019d68fd7e659f5818f51262eb6191048d279/json

**Comment by mlmeyers** [comment]
I am encountering this same issue with my containers starting today. not sure what triggered it and so far unable to resolve. 

Line 635: [19:23:54.397][LocalhostForwarder][Info   ] TransortErrorException 1
	Line 637: [19:23:54.398][LocalhostForwarder][Info   ] Failed to forward tcp port 27017 to 172.31.149.188:27017
	Line 638: [19:23:54.398][LocalhostForwarder][Info   ] Reconciliation failed: invalid argument

**Comment by mlmeyers** [comment]
This is a giant pita. After tons of trial and error I was able to get my machine working again. 

When i encounter the error I also note that the com.docker.localhost-forwarder never spins up the listening port when viewed in processes explorer.

In the end I nuked everything, docker, hyper-v, containers and re-installed. That still did not sort it. 

Finally removing latest Windows 10 update to revert to build version 16299.192 resolved it for me. 

I would like to say this was the fix but I have encountered this as well on another machine running on Windows 10 build 16299.192. On that machine I needed to nuke and re-install everything and that sorted it out. 

With all this I wish I could pinpoint the issue, nuke and pave is far from an ideal solution to this problem. 

I am in process of rolling out in our org, I suspect I will encounter this again and if so will post back with additional findings if any.
